runtime/compiler/aarch64/codegen/ARM64JNILinkage.cpp

```diff
@@ -97,7 +97,7 @@ void J9::ARM64::JNILinkage::releaseVMAccess(TR::Node *callNode, TR::Register *vm
    // Arm Architecture Reference Manual states:
    // "This architecture assumes that all PEs that use the same operating system or hypervisor are in the same Inner Shareable shareability domain"
    // thus, inner shareable dmb suffices
-   generateSynchronizationInstruction(cg(), TR::InstOpCode::dmb, callNode, 0xb);
+   generateSynchronizationInstruction(cg(), TR::InstOpCode::dmb, callNode, TR::InstOpCode::ishst);
```

Comment:
0xb means `TR::InstOpCode::ish`, while the comments above says `dmb ishst`.

Reply:
Would a `dmb ishst` be sufficient for the spin lock implementation here?

Looking at the example spin lock implementation in the Arm Architecture Reference Manual (Appendix K 14.3.1), I wonder if we could replace the `dmb` with a `prfm` and relax the `ldxrx` to a `ldaxrx`.
<img width="653" alt="image" src="https://github.com/user-attachments/assets/1883da8b-fa41-429b-8a63-79599572897c">

Reply:
Actually, is the `dmb` necessary given that we are using `ldxrx` to read the lock?

Reply:
I am trying to understand the memory semantics.
Looking at some other sample code, it seems we don't need the `dmb` instruction.

Reply:
I haven't realized, but the AArch64 JIT code seems to be a kind of old-fashioned.
The traditional 32-bit ARM architecture before ARMv8 did not have the "Load-Acquire" and "Store-Release" instructions, thus needing `DMB` instructions.  `LDA*` and `STL*` instructions for the AArch32 mode were introduced in ARMv8 architecture.

My proposal is to
- Replace `0xb` by `TR::InstOpCode::ish` in this PR so that it does not change the generated code.  Some comments need to be changed, too.
- Open a new issue for improving the code sequence by taking advantage of "Load-Acquire" instructions.

Reply:
I opened Issue #20247 for the future work.